federation/schema.go

```diff
@@ -83,6 +91,47 @@ func convertVersionedSchemas(schemas serviceSchemas) (*SchemaWithFederationInfo,
 	fieldInfos := make(map[*graphql.Field]*FieldInfo)
 	for _, service := range serviceNames {
 		for _, typ := range serviceSchemasByName[service].Schema.Types {
+			
+			// For federated fields parse the arguments to figure out which 
+			// fields are the federated keys. They annotate that information 
+			// on the field object.
+			if typ.Name == "Federation" {
+				for _, field := range typ.Fields {
+					for _, arg := range field.Args {
+						
+						// Extract the type name from the formatting <object>-<service>
+						rootType := getRootType(arg.Type)
+						names := strings.SplitN(field.Name, "-", 2)
+						objName := names[0]
```

Comment:
Why we never use names[1]?

Reply:
Since we know the service we are on, we don't need to use names[1]. We do however need the field to be named "objectName-serviceName" so that when we merge the schemas it gets treated as an independent field func instead of a field collision. This also allows us to query the correct field in the executor.